src/Partials/InputField.vala

```diff
@@ -89,138 +93,41 @@ public class Akira.Partials.InputField : Gtk.EventBox {
-                } else {
-                    entry.primary_icon_name = "";
-                }
-            }
-            return false;
-        });
-
-        leave_notify_event.connect (event => {
-            if (event.detail != Gdk.NotifyType.INFERIOR) {
-                spin_grid.visible = false;
-                spin_grid.no_show_all = true;
-                if (icon_right) {
-                    entry.secondary_icon_name = icon;
-                } else {
-                    entry.primary_icon_name = icon;
-                }
-            }
-            return false;
-        });
-
-        entry.focus_in_event.connect (() => {
-            spin_grid.visible = false;
-            spin_grid.no_show_all = true;
-            if (icon_right) {
-                entry.secondary_icon_name = icon;
-            } else {
-                entry.primary_icon_name = icon;
-            }
-        });
+        add (entry);
+    }
+
+    public void set_range (double min_value, double max_value) {
+        entry.set_range (min_value, max_value);
     }
 
-    private bool handle_key (Gdk.EventKey key) {
+    private bool handle_key_press (Gdk.EventKey key) {
         // Arrow UP
-        if (key.keyval == 65362) {
-            increase_value (key);
+        if (key.keyval == Gdk.Key.Up && key.state == Gdk.ModifierType.SHIFT_MASK) {
+            entry.spin (Gtk.SpinType.STEP_FORWARD, 10);
             return true;
         }
 
         // Arrow DOWN
-        if (key.keyval == 65364) {
-            decrease_value (key);
+        if (key.keyval == Gdk.Key.Down && key.state == Gdk.ModifierType.SHIFT_MASK) {
+            entry.spin (Gtk.SpinType.STEP_BACKWARD, 10);
             return true;
         }
 
         return false;
     }
 
-    public void increase_value (Gdk.EventKey? key) {
-        int num = key != null && key.state.to_string () == "GDK_SHIFT_MASK" ? 10 : 1;
-        double src = double.parse (entry.text) + num;
-        entry.text = src.to_string ();
-    }
-
-    public void decrease_value (Gdk.EventKey? key) {
-        int num = key != null && key.state.to_string () == "GDK_SHIFT_MASK" ? 10 : 1;
-        double src = double.parse (entry.text) - num;
-        entry.text = src.to_string ();
-    }
-
-    public bool button_up_event (Gdk.Event event) {
-        if (event.type == Gdk.EventType.BUTTON_PRESS) {
-            increase_value (null);
+    private bool handle_key_release (Gdk.EventKey key) {
+        if (key.keyval != Gdk.Key.Up && key.keyval != Gdk.Key.Down) {
+            entry.update ();
         }
+
         return false;
     }
 
-    public bool button_down_event (Gdk.Event event) {
-        if (event.type == Gdk.EventType.BUTTON_PRESS) {
-            decrease_value (null);
+    private bool handle_scroll_event (Gdk.EventScroll event) {
+        // If the input field is not focused, don't change the value.
+        if (!entry.has_focus) {
+            GLib.Signal.stop_emission_by_name (entry, "scroll-event");
```

Comment:
You can return `true` here and it will stop the signal emission at this point, preventing the default scroll event handler to run, without explicitly calling `stop_emission_by_name()`.

Reply:
Ah, thanks for the heads up.
If you have time you could create a small PR to fix it 😁

Reply:
Done :smiley:

Reply:
Thank you so much